python/ray/tune/experiment/trial.py

```diff
@@ -636,7 +636,13 @@ def local_experiment_path(self, local_path: str):
             # relative to the old `local_dir`/`logdir`
             for checkpoint in self.get_trial_checkpoints():
                 checkpoint_dir = checkpoint.dir_or_data
-                assert isinstance(checkpoint_dir, str)
+                if not isinstance(checkpoint_dir, str):
+                    logger.warning(
+                        f"No data found in checkpoint for trial {self} and metrics "
+                        f"{checkpoint.metrics}. Skipping."
+                    )
```

Comment:
Maybe we can also include `type(checkpoint_dir)` as a sanity check in case this somehow ends up being a `Dict` or other type.

Reply:
Good point, will add